dockstore-client/src/main/java/io/dockstore/client/cli/nested/AbstractEntryClient.java

```diff
@@ -877,6 +906,160 @@ public void download(final List<String> args) {
+            //?????what should the return code be?????
+            exceptionMessage(e, "Could not get WES section", ENTRY_NOT_FOUND);
+        }
+
+        if (wesUrl == null || wesUrl.isEmpty()) {
+            String wesEndpointUrl = configSubNode.getString("url");
+            if (wesEndpointUrl == null || wesEndpointUrl.isEmpty()) {
+                errorMessage("No WES URL found in config file and no WES URL entered on command line.", CLIENT_ERROR);
+            }
+            out("WES endpoint url is: " + wesEndpointUrl);
+            wesApiClient.setBasePath(wesEndpointUrl);
+        } else {
+            wesApiClient.setBasePath(wesUrl);
+        }
+
+        /**
+         * Setup authentication credentials for the WES URL
+         */
+        //TODO: somehow make this case insensitive?
+        String wesAuthorizationTypeCredentials = configSubNode.getString("authorization", null);
+        String wesAuthorizationType = null;
+        String wesAuthorizationCredentials = null;
+        //out("wes config authorization string is " + wesAuthorizationTypeCredentials);
+        if (wesAuthorizationTypeCredentials != null) {
+            String[] wesAuthorizationTypeCredentialsArray = wesAuthorizationTypeCredentials.split("\\s+");
+            //out("wes authorization credentials array:" + Arrays.toString(wesAuthorizationTypeCredentialsArray));
+            //out("wes auth array length is:" + wesAuthorizationTypeCredentialsArray.length);
+            if (wesAuthorizationTypeCredentialsArray.length > 1) {
+                wesAuthorizationType = wesAuthorizationTypeCredentialsArray[0];
+                //out("WES authorization type:" + wesAuthorizationType.toString());
+
+                wesAuthorizationCredentials = wesAuthorizationTypeCredentialsArray[1];
+                //out("WES authorization credentials:" + wesAuthorizationCredentials.toString());
+
+                if (wesAuthorizationType.equalsIgnoreCase(BEARER)) {
+                    //out("set token to " + wesAuthorizationCredentials);
+                    // TODO: The WES schema should specify the security scheme so we do not need to add a default header
+                    wesApiClient.addDefaultHeader(AUTHORIZATION, BEARER + " " + wesAuthorizationCredentials);
+                    //wesApiClient.setAccessToken(wesAuthorizationCredentials);
+                } else if (wesAuthorizationType.equalsIgnoreCase(BASIC)) {
+                    //wesApiClient.setPassword(wesAuthorizationCredentials);
+                    wesApiClient.addDefaultHeader(AUTHORIZATION, BASIC + " " + wesAuthorizationCredentials);
+                } else {
+                    out("Could not set Authorization header. Unsupported authorization type in config file. "
+                            + "Please use " + BEARER + " or " + BASIC);
+                }
+            }
+        } else {
+            out("Could not set Authorization header. Authorization key not found in config file. "
```

Comment:
This should be phrased as a warning. I believe there were some implementations that didn't have auth and were just using whitelisting. That may have been temporary, but I don't believe there is a requirement for auth, so this should be phrased accordingly. (It's pretty close).